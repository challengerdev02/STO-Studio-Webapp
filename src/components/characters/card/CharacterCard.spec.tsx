import { cleanup, render } from '@testing-library/react';
import { CharacterCard } from './index';
import React from 'react';
import Img from '../../../../public/sellScene.png';

describe('CharacterCard', function () {
  afterEach(cleanup);

  const setup = () => {
    const utils = render(
      <CharacterCard
        description={'description'}
        artists={'spider-man'}
        title={'title'}
        img={Img}
        onRevise={jest.fn()}
      />
    );
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should test that the card is rendered', function () {
    const { container } = setup();
    expect(container).toBeDefined();
  });

  it('should return the character string in the component correctly', function () {
    const { getByText } = setup();
    expect(getByText('spider-man')).toBeInTheDocument();
  });
});
