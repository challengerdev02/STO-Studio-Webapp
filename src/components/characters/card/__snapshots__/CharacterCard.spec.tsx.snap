// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterCard should match with snapshot 1`] = `
<div>
  <div
    style="cursor: pointer;"
  >
    <div
      class="ant-card sc-bdvvtL cFSdnJ"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="sc-gsDKAQ dBlFEh"
        >
          <img
            alt="img"
            class="sc-dkPtRN edONvH"
            src="/test-file-stub"
          />
        </div>
        <div
          class="sc-gKclnd eIvoRM"
        >
          <div
            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-multiple-line"
          >
            <h5
              class="ant-typography"
            >
              title
            </h5>
          </div>
          <div
            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-multiple-line"
          >
            <span
              class="sc-eCImPb apvtg"
            >
              Description: 
            </span>
            description
          </div>
          <div
            class="ant-typography ant-typography-ellipsis ant-typography-ellipsis-multiple-line"
          >
            <span
              class="sc-eCImPb apvtg"
            >
              Artists: 
            </span>
            spider-man
          </div>
          <button
            class="ant-btn ant-btn-default sc-hKwDye buGTcB"
            type="button"
          >
            <span
              class="sc-jRQBWg iJuCHL"
            >
              Sell Character
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
